### PR TITLE
[Merged by Bors] - feat: to_additive raises linter errors; nested to_additive

### PR DESCRIPTION
**Pull request by fpvandoorn**
* Turn info messages of `to_additive` into linter errors
* Allow `@[to_additive (attr := to_additive)]` to additivize the generated lemma. This is useful for `Pow -> SMul -> VAdd` lemmas. We can write e.g. `@[to_additive (attr := to_additive, simp)]` to add the `simp` attribute to all 3 generated lemmas, and we can provide other options to each `to_additive` call separately (specifying a name / reorder).
* The previous point was needed to cleanly get rid of some linter warnings. It also required some additional changes (`addToAdditiveAttr` now returns a value, turn a few (meta) definitions into `mutual partial def`, reorder some definitions, generalize `additivizeLemmas` to lists of more than 2 elements) that should have no visible effects for the user.

---
- [x] depends on: #1816 [optional extra text]

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1816~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by fpvandoorn** [comment]
note: please delegate, not merge this PR. It likely raises errors when merging master

File: Mathlib/Tactic/ToAdditive.lean [hunk 1]
```diff
@@ -15,6 +15,7 @@ import Std.Data.Option.Basic
 import Std.Tactic.NormCast.Ext -- just to copy the attribute
 import Std.Tactic.Ext.Attr -- just to copy the attribute
 import Std.Tactic.Lint.Simp -- for DiscrTree.elements
+import Std.Tactic.Lint.Frontend -- useful to lint this file
```
Comment (gebner) [inline Mathlib/Tactic/ToAdditive.lean:18]:
```suggestion
import Std.Tactic.Lint -- useful to lint this file
```

**Review by gebner — COMMENTED** [review]
bors d+

I'm not sure I'm a fan of using `logLinter` in functions that are not linters.

**Comment by gebner** [comment]
> I'm happy to use something else. This specifically was requested by @digama0 [here](https://github.com/leanprover-community/mathlib4/pull/1314#discussion_r1063922378).

@digama0 Do you plan to make all warnings disableable?

**Comment by fpvandoorn** [comment]
bors merge

I'm happy to change this to something else in a future PR, if we reach a consensus.

**Pull request closed without merging**